src/Buildalyzer/Environment/BuildEnvironment.cs

```diff
@@ -107,16 +107,12 @@ public sealed class BuildEnvironment
                 _globalProperties.Add(MsBuildProperties.DesignTimeBuild, "true");
                 _globalProperties.Add(MsBuildProperties.BuildingProject, "false"); // Supports Framework projects: https://github.com/dotnet/project-system/blob/main/docs/design-time-builds.md#determining-whether-a-target-is-running-in-a-design-time-build
                 _globalProperties.Add(MsBuildProperties.BuildProjectReferences, "false");
-                _globalProperties.Add(MsBuildProperties.SkipCompilerExecution, "true");

```

Comment:
The changes here are really my only concern. It appears as though, in the interest of supporting Functions apps, we're essentially disabling design-time builds by turning off several of the build props that would otherwise put MSBuild into that mode. In particular, turning compiler execution back on will slow down _every_ use of Buildalyzer. Further down, turning copying to the output directory back on might risk a regression of some problems folks used to have with their built assemblies being overwritten during Buildalyzer use.

I guess in general, I'm not super comfortable changing the global props used for the default behavior since those have evolved over a long timespan. I'm wondering if there's a way to identify an Azure Functions project and make these changes, _but only_ if the project is a Functions app?